cpp/posix.h

```diff
@@ -81,11 +81,15 @@ inline int close(int fd) {
 }
 
 inline int putenv(Str* name, Str* value) {
-  int env_string_size =
-      name->len_ + value->len_ + 1;  // NOTE(Jesse): +1 for the '=' between them
-  char* env_string = static_cast<char*>(malloc(env_string_size));
-  snprintf(env_string, env_string_size, "%s=%s", name->data_, value->data_);
-  return ::putenv(env_string);
+  // NOTE(Jesse): Technically this over-allocates by 1 byte, but by doing that
+  // we don't rely on the underlying Str* data_ buffer being defined as (Str::len_ + 1)
+  int env_string_size = name->len_ + value->len_ + 2;
+
+  Str* env_string = mylib::BlankStr(env_string_size);
```

Comment:
It's in flux right now, but when we switch to GC, I think we will actually use malloc /free here.

Because then you don't need to worry about the buffer moving out from under `putenv()`!

That can only happen on allocations.  (And BTW I want to rename BlankStr to AllocStr to make that very clear.  Also AllocList, AllocDict, etc.)